### PR TITLE
From Vagrant to Docker

### PR DESCRIPTION
**Pull request by simonbnrd**
The following PR allows us to get rid of Vagrant and use Docker + Docker-Compose instead as a development environment. Please note two things I had to change to complete that task:

- fix an encoding issue in `lib/optimus_prime/sources/appsflyer.rb` that @Anumodana, @iammchat and I encountered when running rspec on a specific spec file.

CircleCI will automatically push the new image on the Docker hub. Since the gem is public, the Docker image is also public.

Please have a look at the README to follow the instructions, if something isn't clear enough I'll update it.

### CODE REVIEW

**Comment by jorin-vogel** [comment]
:+1: nice work!

File: README.md [hunk 1]
```diff
@@ -102,17 +102,16 @@ data source, you can use it to make sure that the last chunk gets uploaded.
 
 #### Development
 
-1. Make sure you have [Vagrant](vagrant) installed.
-2. Run `vagrant up` to start and setup the VM.
-3. Run `vagrant ssh` to login to the server.
-4. You can run the tests with `rake`.
+1. You need [Docker][install-docker] and [docker-compose][install-compose].
+To install them on OSX just run `brew install docker boot2docker docker-compose`. To start the boot2docker VM you need to run `boot2docker init && boot2docker up` and make sure you set the environment variables as described by the `boot2docker up` command. Feel free to use the `source <(boot2docker shellinit)` command to write and load the _pem_ files required by Docker. The included `docker-compose.yml` file describes all containers needed to run the application in development. For now it only uses an additional _data_ container that uses the same image as the main application. This container exposes `/home/playlab/ruby` as a volume, which is then included in the main app container using the `volume_from` directive. This allows us to persist any gems installed in the development container.
```
Comment (omarkhan) [inline README.md]: Can we wrap this at 80 chars?

**Comment by omarkhan** [comment]
Looks great! Nice work :+1:

File: README.md (hunk 1 above)
Comment (simonbnrd) [inline README.md]: Sure, I'm polishing everything right now and I'm doing the same for Bumblebee.

**Comment by T-Dnzt** [comment]
Nice, nice :+1: Gonna miss Vagrant! (or not)

**Comment by Anumodana** [comment]
Great! :smiley: :+1:

**Comment by omarkhan** [comment]
Very nice. Let's get this merged

**Comment by omarkhan** [comment]
Why do we need to include the Gemfile.lock? I thought that wasn't the recommended way for gems

**Comment by simonbnrd** [comment]
@omarkhan : Yep I'll remove it, for a time I had to copy it because of a permission issue after a `bundle install`. And like datacubes, there's no need to save gems in a version-specific directory like it's actually done in the Dockerfile, I'll also remove that. Commit coming soon.

**Comment by simonbnrd** [comment]
After a talk with Kevin Bell from CircleCI tonight I had more informations about the issues I had with Docker on CircleCI. This [PR](https://github.com/docker/docker/pull/9635) landed until 1.5, and CircleCI is still on Docker 1.4. Kevin told me they usually try to upgrade services faster than that, but Docker is a pretty significant thing for them to upgrade. It's just that relative paths weren't supported for COPY until a later version of Docker. So, even though docker knows about the WORKDIR, and would respect it for running commands, etc, it doesn't respect it for COPY (in the 1.4 version of docker at least).

Btw for now I've disabled the deployment of the Docker image on our hub registry, the reason is that the whole build takes already a lot of time compared to before because of Docker, I tried to cut off time consuming steps as much as possible.

**Comment by jorin-vogel** [comment]
Nice work @simonbnrd!

**Comment by simonbnrd** [comment]
I've spotted another issue on CircleCI, I'm waiting for an answer from the staff before merging the branch.

**Comment by simonbnrd** [comment]
For now I've made the choice to not redirect `docker-compose build` to `/dev/null` like it's done in datacubes or passport. Since first builds can take a long time it's better for people to have an indicator of the building process and clearly notice when Docker's cache is used or not.

**Pull request merged**